src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -0,0 +1,523 @@
+          </tr>
+          <tr>
+              <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> @ref Utils_AccurateMassSearch </td>
+          </tr>
+      </table>
+  </CENTER>
+
+  Build an assay library from DDA data (MS and MS/MS) (mzML).
+  Please provide a list of features found in the data (featureXML).
+
+  Features can be detected using the FeatureFinderMetabo (FFM) and identifcation information
+  can be applied using the AccurateMassSearch feautreXML output.
+
+  Note: Please set the "report_confex_hulls" parameter to "true" when using the FFM.
+
+  If the FFM featureXML is used the "use_known_unknowns" flag is used automatically.
+
+  Internal procedure AssayGeneratorMetabo:
+  1. Input mzML and featureXML
+  2. Annotate precursor mz and intensity
+  3. Filter feature by convexhull size
+  4. Assign precursors to specific feature
+  5. Extract feature meta information (if possible)
+  6. Find MS2 spectrum with highest intensity precursor for one feature
+  7. Dependent on the method use the MS2 with the highest intensity precursor or a consensus spectrum
+     for the transition calculation
+  8. Calculate thresholds (maximum and minimum intensity for transition peak)
+  9. Extract and write transitions
+
+  <B>The command line parameters of this tool are:</B>
+  @verbinclude UTILS_SiriusAdapter.cli
+  <B>INI file documentation of this tool:</B>
+  @htmlinclude UTILS_SiriusAdapter.html
+ */
+
+/// @cond TOPPCLASSES
+
+
+/// struct to hold assay information of one row 
+struct AssayRow
+{
+  double precursor_mz;
+  double product_mz;
+  float library_int;
+  double normalized_rt;
+  String compound_name;
+  String smiles;
+  String sumformula;
+  String adduct;
+  String transition_group_id;
+  String transition_id;
+  bool decoy; 
+};
+
+class TOPPAssayGeneratorMetabo :
+  public TOPPBase
+{
+public:
+  TOPPAssayGeneratorMetabo() :
+    TOPPBase("AssayGeneratorMetabo", "Assay library generation from DDA data (Metabolomics)", false)
+    {}
+
+protected:
+
+  void registerOptionsAndFlags_() override
+  {
+
+    registerInputFile_("in", "<file>", "", "MzML Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzml"));
+
+    registerInputFile_("in_id", "<file>", "", "FeatureXML Input with id information (accurate mass search)");
+    setValidFormats_("in_id", ListUtils::create<String>("featurexml"));
+
+    registerOutputFile_("out", "<file>", "", "Assay library output file");
+    setValidFormats_("out", ListUtils::create<String>("tsv"));
+
+    registerStringOption_("method", "<choice>", "highest_intensity", "",false);
+    setValidStrings_("method", ListUtils::create<String>("highest_intensity,consensus_spectrum"));
+
+    registerDoubleOption_("precursor_mz_tolerance", "<num>", 0.005, "Tolerance window (left and right) for precursor selection", false);
+    registerDoubleOption_("precursor_rt_tolerance", "<num>", 5, "Tolerance window (left and right) for precursor selection", false);
```

Comment:
it would be nice to have those in `ppm` and to in seconds (is it already or is it not?)

Reply:
will add ppm and rt is in seconds